src/main/java/com/amplitude/Amplitude.java

```diff
@@ -137,7 +139,11 @@ public synchronized void flushEvents() {
       eventsToSend.clear();
       CompletableFuture.supplyAsync(
           () -> {
-            Response response = httpCall.syncHttpCallWithEventsBuffer(eventsInTransit);
+            Response response = null;
+            try {
+              response = httpCall.syncHttpCallWithEventsBuffer(eventsInTransit);
+            } catch (AmplitudeInvalidAPIKeyException e) {
+            }
```

Comment:
I think in general, the flow should be to stop the whole Amplitude SDK on any invalid API key. Thoughts?

Reply:
User was specifically asking, and I agree, for a hard noticeable stop in the program when the API key is wrong or missing.

Reply:
yeah, ideally the flow should stop and we are not sending anything further.